File: csharp/Behavioral/13-ChainOfResponsibility/Handler/SupervisorApprover.cs
public class SupervisorApprover : BaseApprovalHandler
{
    private const decimal Limit = 5_000m;

    public override void Handle(ApprovalRequest request)
    {
        if (request.Cost <= Limit)
            Console.WriteLine($"  ✅ Supervisor approved: {request}");
        else
        {
            Console.WriteLine($"  ↗️  Supervisor escalating (N${request.Cost:N2} > N${Limit:N2})...");
            base.Handle(request);
        }
    }
}

File: typescript/Behavioral/13-ChainOfResponsibility/Handler/SupervisorApprover.ts
import { BaseApprovalHandler } from "./BaseApprovalHandler";
import { ApprovalRequest } from "../Model/ApprovalRequest";

export class SupervisorApprover extends BaseApprovalHandler {
  private static readonly LIMIT = 5_000;

  handle(request: ApprovalRequest): void {
    if (request.cost <= SupervisorApprover.LIMIT)
      console.log(`  ✅ Supervisor approved: ${request}`);
    else {
      console.log(
        `  ↗️  Supervisor escalating (N$${request.cost.toFixed(2)} > N$${SupervisorApprover.LIMIT.toFixed(2)})...`,
      );
      super.handle(request);
    }
  }
}
